src/Uno.UI.Tests/Windows_UI_Xaml_Data/BindingHelperTests/Given_XamlBindingHelper.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Markup;
+
+namespace Uno.UI.Tests.Windows_UI_Xaml_Data.CollectionViewTests
+{
+	[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
+	public class Given_XamlBindingHelper
+	{
+		[TestMethod]
+		[DataRow("42", (int)42)]
+		[DataRow("42", (double)42)]
+		[DataRow("42.42", (double)42.42)]
+		public void When_StringToNumber(string value, object expected)
+		{
+			Assert.AreEqual(expected, XamlBindingHelper.ConvertValue(expected.GetType(), value));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'expected' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4930614)